torch/distributed/elastic/rendezvous/api.py

```diff
@@ -83,7 +83,14 @@ def build(
         # TODO swap to collectives comms API
         if rank == 0:
             addr = local_addr or socket.getfqdn()
-            port = _get_free_port()
+            # If the store is not a TCPStore, we need to create a new TCPStore.
+            # Also to avoid race condition in getting port for TCPStore, we will reuse
+            # the existing store as much as possible.
+            if hasattr(store, "port"):
+                port = store.port
+            else:
+                # Not all stores define the property of port, so we fallback to get_free_port.
+                port = get_free_port()
```

Comment:
Not sure if we want to create a TCPStore here directly.

Reply:
I'm not sure this change is safe -- we may want to change the build() interface to pass in an optional port

If we're not using a shared store we shouldn't be passing the shared store port to the child workers. And the shared store path bypasses this function I believe and just directly creates a RendezvousStoreInfo object